lib/mayaUsd/fileio/shading/shadingModeExporterContext.cpp

```diff
@@ -440,111 +451,246 @@ UsdMayaShadingModeExportContext::MakeStandardMaterialPrim(
+        }
     }
 
-    return UsdPrim();
-}
+    const UsdShadeMaterial& getMaterial(const TfToken& shapeName)
+    {
+        // Look for an existing material for the requested shape:
+        const NameVec&                   uvNames = _shapeNameToUVNames[shapeName];
+        MaterialMappings::const_iterator iter = _uvNamesToMaterial.find(uvNames);
+        if (iter != _uvNamesToMaterial.end()) {
+            return iter->second;
+        }
 
+        // Create a specialized material:
+        std::string newName = _material.GetPrim().GetName();
+        for (const TfToken& t : uvNames) {
+            newName += "_";
+            newName += t.GetString();
+        }
+        SdfPath newPath
+            = _material.GetPrim().GetPath().GetParentPath().AppendChild(TfToken(newName.c_str()));
+        UsdShadeMaterial newMaterial
+            = UsdShadeMaterial::Define(_material.GetPrim().GetStage(), newPath);
+        newMaterial.GetPrim().GetSpecializes().AddSpecialize(_material.GetPrim().GetPath());
+
+        NameVec::const_iterator itNode = _nodesWithUVInput.cbegin();
+        NameVec::const_iterator itName = uvNames.cbegin();
+        for (; itNode != _nodesWithUVInput.cend(); ++itNode, ++itName) {
+            TfToken inputName(
+                TfStringPrintf("%s:%s", itNode->GetText(), _tokens->varname.GetText()));
+            UsdShadeInput materialInput
+                = newMaterial.CreateInput(inputName, SdfValueTypeNames->Token);
+            materialInput.Set(*itName);
+        }
+        auto insertResult
+            = _uvNamesToMaterial.insert(MaterialMappings::value_type { uvNames, newMaterial });
+        return insertResult.first->second;
+    }
+
+private:
+    /// The original material:
+    const UsdShadeMaterial& _material;
+    /// Helper structures for UV set mappings:
+    using NameVec = std::vector<TfToken>;
+    NameVec _nodesWithUVInput;
+    using ShapeToStreams = std::map<TfToken, NameVec>;
+    ShapeToStreams _shapeNameToUVNames;
+    using MaterialMappings = std::map<NameVec, UsdShadeMaterial>;
+    MaterialMappings _uvNamesToMaterial;
+};
+} // namespace
+
+void UsdMayaShadingModeExportContext::BindStandardMaterialPrim(
+    const UsdPrim&          materialPrim,
+    const AssignmentVector& assignmentsToBind,
+    SdfPathSet* const       boundPrimPaths) const
+{
+    UsdShadeMaterial material(materialPrim);
+    if (!material) {
+        TF_RUNTIME_ERROR("Invalid material prim.");
+        return;
+    }
+
+    _UVMappingManager uvMappingManager(material, assignmentsToBind);
+
+    UsdStageRefPtr stage = GetUsdStage();
+    TfToken        materialNameToken(materialPrim.GetName());
+    for (const auto& iter : assignmentsToBind) {
+        const SdfPath&    boundPrimPath = iter.boundPrimPath;
+        const VtIntArray& faceIndices = iter.faceIndices;
+        const TfToken&    shapeName = iter.shapeName;
+
+        const UsdShadeMaterial& materialToBind = uvMappingManager.getMaterial(shapeName);
```

Comment:
Main difference in the extracted code for binding materials is here. We ask the mapping manager for the material to bind. In case of inconsistent texcoord naming in the shape set, this function can return specialized materials that fit the needs of the current shape.